src/main/java/org/javarosa/xpath/expr/XPathDecryptStringFunc.java

```diff
@@ -35,19 +37,20 @@ public Object evalBody(DataInstance model, EvaluationContext evalContext, Object
      * @param o2 the key used for encryption
      * @param o3 the encryption algorithm to use
      */
-    private static String decryptString(Object o1, Object o2, Object o3) {
+    private String decryptString(Object o1, Object o2, Object o3) {
         String message = FunctionUtils.toString(o1);
         String key = FunctionUtils.toString(o2);
         String algorithm = FunctionUtils.toString(o3);
 
-        if (!algorithm.equals("AES")) {
+        if (!algorithm.equals(CC_KEY_ALGORITHM_AES) &&
+                !algorithm.equals(CC_KEY_ALGORITHM_RSA)) {
             throw new XPathException("Unknown algorithm \"" + algorithm +
                     "\" for " + NAME);
         }
 
         try {
-            return decrypt(message, key);
-        } catch (EncryptionUtils.EncryptionException e) {
+            return encryptionHelper.decrypt(algorithm, message, key);
```

Comment:
ah, I see, I don't think this is actually the right way for this to work. EncryptionHelper is playing way too many roles in this context. I'm very worried that things like the existence of a KeyStore are coming into play under the hood of calls like this, which are dangerous to be mixing together. 

I don't see a specific reason why it makes sense for this class's implementation to switch from static to object oriented, and there are meaningful runtime performance implementation differences from coupling new lifecycle objects into this code.